### PR TITLE
feat(network): Close temporary stream connections

### PR DESCRIPTION
**Pull request by juslesan**
## Summary

The TemporaryConnections were not properly closed directly after use. Instead the implementation waited for the connections to be eventually closed by the connection manager garbage collector.

The temorary connections are now closed after use with a new rpc call.

### CODE REVIEW

File: packages/trackerless-network/src/logic/RandomGraphNode.ts [hunk 1]
```diff
@@ -308,7 +311,7 @@ export class RandomGraphNode extends EventEmitter<Events> {
             markAndCheckDuplicate(this.duplicateDetectors, msg.messageId!, msg.previousMessageRef)
         }
         this.emit('message', msg)
-        this.config.propagation.feedUnseenMessage(msg, this.getPropagationTargets(msg), previousNode ?? null)
+        this.config.propagation.feedUnseenMessage(msg, this.getPropagationTargets(msg), null)
```
Comment (harbu) [inline packages/trackerless-network/src/logic/RandomGraphNode.ts]: Is this change intended to be included?

Comment (teogeb) [inline packages/trackerless-network/src/logic/RandomGraphNode.ts]: Is this change part of a separate functionality change? Or how this relates to the closing temporary connections feature?

File: packages/trackerless-network/test/unit/TemporaryConnectionRpcLocal.test.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    })
+
+    afterEach(() => {
+        rpcCommunicator.destroy()
+    })
+
+    it('Open Connection', async () => {
+        const caller = createMockPeerDescriptor()
+        await rpcLocal.openConnection(peerDescriptor, { incomingSourceDescriptor: caller } as any)
```
Comment (harbu) [inline packages/trackerless-network/test/unit/TemporaryConnectionRpcLocal.test.ts]: Should the 1st argument be instance of `TemporaryConnectionRequest` (de facto: unused)?

**Review by harbu — CHANGES_REQUESTED** [review]

File: packages/trackerless-network/test/unit/TemporaryConnectionRpcLocal.test.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    let rpcCommunicator: ListeningRpcCommunicator
+    let rpcLocal: TemporaryConnectionRpcLocal
+
+    beforeEach(() => {
+        rpcCommunicator = new ListeningRpcCommunicator('mock', new MockTransport())
+        rpcLocal = new TemporaryConnectionRpcLocal({
+            localPeerDescriptor: peerDescriptor,
+            rpcCommunicator
+        })
+
```
Comment (teogeb) [inline packages/trackerless-network/test/unit/TemporaryConnectionRpcLocal.test.ts]:
```suggestion
```

File: packages/trackerless-network/test/unit/TemporaryConnectionRpcLocal.test.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    })
+
+    afterEach(() => {
+        rpcCommunicator.destroy()
+    })
+
+    it('Open Connection', async () => {
```
Comment (teogeb) [inline packages/trackerless-network/test/unit/TemporaryConnectionRpcLocal.test.ts]: Maybe redundant test case

**Review by teogeb — COMMENTED** [review]
First round of comments

**Review by harbu — APPROVED** [review]
LGTM; I think @teogeb might've still had some points, let's wait for him to get back to this?

**Review by teogeb — APPROVED** [review]

**Pull request merged**